src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -613,6 +613,27 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             }
             break;
 
+            case NI_Vector64_GetElement:
+            case NI_Vector128_GetElement:
+            case NI_Vector64_ToScalar:
+            case NI_Vector128_ToScalar:
+            {
+                ssize_t indexValue = 0;
+                if (intrin.id == NI_Vector64_GetElement || intrin.id == NI_Vector128_GetElement)
+                {
+                    assert(intrin.op2->IsCnsIntOrI());
+                    indexValue = intrin.op2->AsIntCon()->gtIconVal;
+                }
+
+                // no-op if vector is float/double, targetReg == op1Reg and fetching for 0th index.
+                if (!(varTypeIsFloating(intrin.baseType) && targetReg == op1Reg && indexValue == 0))
```

Comment:
I find the following more readable...
```suggestion
                if (varTypeIsIntegral(intrin.baseType) || ((targetReg == op1Reg) && (indexValue == 0)))
```

Reply:
we could also just do:
```
if (varTypeIsFloating(intrin.baseType) && (targetReg == op1Reg) && (indexValue == 0))
{
    break;
}
```